### ISSUE TITLE
File descriptor leak when failing to connect to Redis

### ISSUE THREAD

**Issue by kilgor** [slact/nchan#411]
Hi, 
nginx.conf 
`nchan_redis_url "redis://127.0.0.1:6379";`
When the redis service disable after nginx started i get nginx errro message in log
`[error] 32047#32047: nchan: Can't reconnect to redis at redis://127.0.0.1:6389: Connection refused.`
this is ok. But each 5 seconds nginx tries to recconnect and leaves socket connection open.

```
ps -aux | grep nginx
root     28754  0.0  0.0 344932  7564 ?        SNs  10:20   0:00 nginx: master process /usr/sbin/nginx -g daemon on; master_process on;
www-data 28755  0.0  0.0 345712 15140 ?        SN   10:20   0:03 nginx: worker process
www-data 28756  0.0  0.0 346088 15488 ?        SN   10:20   0:07 nginx: worker process
```

```
lsof of nginx worker.
lsof -p 28756
nginx   28756 www-data 5247u  sock                0,8      0t0 1406775427 can't identify protocol
nginx   28756 www-data 5248u  sock                0,8      0t0 1406775428 can't identify protocol
nginx   28756 www-data 5249u  sock                0,8      0t0 1406775882 can't identify protocol
nginx   28756 www-data 5250u  sock                0,8      0t0 1406775883 can't identify protocol
nginx   28756 www-data 5251u  sock                0,8      0t0 1406775933 can't identify protocol
nginx   28756 www-data 5252u  sock                0,8      0t0 1406775934 can't identify protocol
nginx   28756 www-data 5253u  sock                0,8      0t0 1406776005 can't identify protocol
nginx   28756 www-data 5254u  sock                0,8      0t0 1406776006 can't identify protocol
nginx   28756 www-data 5255u  sock                0,8      0t0 1406776038 can't identify protocol
nginx   28756 www-data 5256u  sock                0,8      0t0 1406776039 can't identify protocol
```
and many similar

```
lsof -p 28756 | wc -l
5313

```

Finally whole nginx hangs with 
`[crit] 32047#32047: accept4() failed (24: Too many open files)`

```
Limit                     Soft Limit           Hard Limit           Units
Max cpu time              unlimited            unlimited            seconds
Max file size             unlimited            unlimited            bytes
Max data size             unlimited            unlimited            bytes
Max stack size            10485760             10485760             bytes
Max core file size        0                    0                    bytes
Max resident set          unlimited            unlimited            bytes
Max processes             unlimited            unlimited            processes
Max open files            50000                50000                files
Max locked memory         65536                65536                bytes
Max address space         unlimited            unlimited            bytes
Max file locks            unlimited            unlimited            locks
Max pending signals       62793                62793                signals
Max msgqueue size         819200               819200               bytes
Max nice priority         0                    0
Max realtime priority     0                    0
Max realtime timeout      unlimited            unlimited            us
```

**Comment by slact** [comment]
Fixed in 7abc33b5 -- there was some legacy code that didn't belong there anymore. Thanks for the detailed bug report.

**Comment by kilgor** [comment]
works like a charm. thanks
